Modules/_operator.c

```diff
@@ -967,6 +969,16 @@ itemgetter_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     Py_INCREF(item);
     ig->item = item;
     ig->nitems = nitems;
+    ig->index = -1;
+    if (PyLong_CheckExact(item)) {
+        index = PyLong_AsSsize_t(item);
+        if (index < 0) {
+            // Only optimize non-negative values that fit in a Py_ssize_t
```

Comment:
Maybe add `assert(PyErr_ExceptionMatches(PyExc_OverflowError))` just for a case?

Reply:
That doesn't make sense to me.  The *OverflowError* is only happening when `index == -1`; otherwise, a negative index just means a negative input.